### PR TITLE
Threading issues #7815

### PR DESCRIPTION
**Pull request by rymsha**

### CODE REVIEW

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/core/core-cluster/src/test/java/com/enonic/xp/cluster/impl/ClusterCheckSchedulerImplTest.java  1
- modules/core/core-internal/src/main/java/com/enonic/xp/core/internal/concurrent/ManagedRecurringJobScheduler.java  1
- modules/server/server-udc/src/main/java/com/enonic/xp/server/udc/impl/UdcActivator.java  2
- modules/server/server-trace/src/main/java/com/enonic/xp/server/internal/trace/event/TraceEventDispatcherExecutorImpl.java  1
- modules/core/core-app/src/main/java/com/enonic/xp/core/impl/app/ApplicationListenerHubExecutorImpl.java  1
- modules/core/core-event/src/main/java/com/enonic/xp/core/impl/event/EventPublisherExecutorImpl.java  1
- modules/core/core-cluster/src/main/java/com/enonic/xp/cluster/impl/ClusterCheckSchedulerImpl.java  1
- modules/core/core-internal/src/main/java/com/enonic/xp/core/internal/concurrent/ThreadFactoryImpl.java  2
- modules/server/server-udc/src/main/java/com/enonic/xp/server/udc/impl/UdcSchedulerImpl.java  1
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/issue/IssueMailSendExecutorImpl.java  1
- modules/core/core-internal/src/main/java/com/enonic/xp/core/internal/concurrent/ManagedExecutor.java  3
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/TaskManagerCleanupSchedulerImpl.java  1
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/TaskManagerExecutorImpl.java  1
         

Complexity decreasing per file
==============================
+ modules/core/core-cluster/src/test/java/com/enonic/xp/cluster/impl/ClusterManagerImplTest.java  -1
+ modules/server/server-udc/src/main/java/com/enonic/xp/server/udc/impl/UdcService.java  -1
+ modules/core/core-event/src/main/java/com/enonic/xp/core/impl/event/EventPublisherImpl.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=4812607&bid=16006411)

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/TaskManagerCleanupSchedulerImpl.java  1
- modules/core/core-internal/src/main/java/com/enonic/xp/core/internal/concurrent/ManagedRecurringJobScheduler.java  1
- modules/server/server-udc/src/main/java/com/enonic/xp/server/udc/impl/UdcSchedulerImpl.java  1
- modules/core/core-internal/src/main/java/com/enonic/xp/core/internal/concurrent/ManagedExecutor.java  3
- modules/core/core-app/src/main/java/com/enonic/xp/core/impl/app/ApplicationListenerHubExecutorImpl.java  1
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/issue/IssueMailSendExecutorImpl.java  1
- modules/core/core-event/src/main/java/com/enonic/xp/core/impl/event/EventPublisherExecutorImpl.java  1
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/TaskManagerExecutorImpl.java  1
- modules/server/server-trace/src/main/java/com/enonic/xp/server/internal/trace/event/TraceEventDispatcherExecutorImpl.java  1
- modules/core/core-cluster/src/test/java/com/enonic/xp/cluster/impl/ClusterCheckSchedulerImplTest.java  1
- modules/server/server-udc/src/main/java/com/enonic/xp/server/udc/impl/UdcActivator.java  2
- modules/core/core-cluster/src/main/java/com/enonic/xp/cluster/impl/ClusterCheckSchedulerImpl.java  1
- modules/core/core-internal/src/main/java/com/enonic/xp/core/internal/concurrent/ThreadFactoryImpl.java  2
         

Complexity decreasing per file
==============================
+ modules/core/core-cluster/src/test/java/com/enonic/xp/cluster/impl/ClusterManagerImplTest.java  -1
+ modules/server/server-udc/src/main/java/com/enonic/xp/server/udc/impl/UdcService.java  -1
+ modules/core/core-event/src/main/java/com/enonic/xp/core/impl/event/EventPublisherImpl.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=4812607&bid=16006411)

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/core/core-cluster/src/test/java/com/enonic/xp/cluster/impl/ClusterCheckSchedulerImplTest.java  1
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/issue/IssueMailSendExecutorImpl.java  1
- modules/core/core-internal/src/main/java/com/enonic/xp/core/internal/concurrent/ThreadFactoryImpl.java  2
- modules/core/core-internal/src/main/java/com/enonic/xp/core/internal/concurrent/ManagedExecutor.java  3
- modules/core/core-app/src/main/java/com/enonic/xp/core/impl/app/ApplicationListenerHubExecutorImpl.java  1
- modules/core/core-cluster/src/main/java/com/enonic/xp/cluster/impl/ClusterCheckSchedulerImpl.java  1
- modules/core/core-internal/src/main/java/com/enonic/xp/core/internal/concurrent/ManagedRecurringJobScheduler.java  4
- modules/server/server-udc/src/main/java/com/enonic/xp/server/udc/impl/UdcActivator.java  2
- modules/core/core-internal/src/test/java/com/enonic/xp/core/internal/concurrent/ManagedRecurringJobSchedulerTest.java  1
- modules/core/core-event/src/main/java/com/enonic/xp/core/impl/event/EventPublisherExecutorImpl.java  1
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/TaskManagerCleanupSchedulerImpl.java  1
- modules/server/server-udc/src/main/java/com/enonic/xp/server/udc/impl/UdcSchedulerImpl.java  1
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/TaskManagerExecutorImpl.java  1
- modules/server/server-trace/src/main/java/com/enonic/xp/server/internal/trace/event/TraceEventDispatcherExecutorImpl.java  1
         

Complexity decreasing per file
==============================
+ modules/core/core-event/src/main/java/com/enonic/xp/core/impl/event/EventPublisherImpl.java  -1
+ modules/core/core-cluster/src/test/java/com/enonic/xp/cluster/impl/ClusterManagerImplTest.java  -1
+ modules/server/server-udc/src/main/java/com/enonic/xp/server/udc/impl/UdcService.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=4812607&bid=16006411)

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/server/server-trace/src/main/java/com/enonic/xp/server/internal/trace/event/TraceEventDispatcherExecutorImpl.java  1
- modules/core/core-app/src/main/java/com/enonic/xp/core/impl/app/ApplicationListenerHubExecutorImpl.java  1
- modules/core/core-internal/src/test/java/com/enonic/xp/core/internal/concurrent/ManagedRecurringJobSchedulerTest.java  1
- modules/core/core-internal/src/test/java/com/enonic/xp/core/internal/concurrent/ManagedExecutorTest.java  1
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/TaskManagerCleanupSchedulerImpl.java  1
- modules/core/core-event/src/main/java/com/enonic/xp/core/impl/event/EventPublisherExecutorImpl.java  1
- modules/core/core-internal/src/main/java/com/enonic/xp/core/internal/concurrent/ManagedExecutor.java  3
- modules/server/server-udc/src/main/java/com/enonic/xp/server/udc/impl/UdcSchedulerImpl.java  1
- modules/server/server-udc/src/main/java/com/enonic/xp/server/udc/impl/UdcActivator.java  2
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/issue/IssueMailSendExecutorImpl.java  1
- modules/core/core-cluster/src/test/java/com/enonic/xp/cluster/impl/ClusterCheckSchedulerImplTest.java  1
- modules/core/core-internal/src/main/java/com/enonic/xp/core/internal/concurrent/ThreadFactoryImpl.java  2
- modules/core/core-internal/src/main/java/com/enonic/xp/core/internal/concurrent/ManagedRecurringJobScheduler.java  4
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/TaskManagerExecutorImpl.java  1
- modules/core/core-cluster/src/main/java/com/enonic/xp/cluster/impl/ClusterCheckSchedulerImpl.java  1
         

Complexity decreasing per file
==============================
+ modules/core/core-cluster/src/test/java/com/enonic/xp/cluster/impl/ClusterManagerImplTest.java  -1
+ modules/server/server-udc/src/main/java/com/enonic/xp/server/udc/impl/UdcService.java  -1
+ modules/core/core-event/src/main/java/com/enonic/xp/core/impl/event/EventPublisherImpl.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=4812607&bid=16006411)

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/core/core-internal/src/main/java/com/enonic/xp/core/internal/concurrent/ThreadFactoryImpl.java  2
- modules/server/server-udc/src/main/java/com/enonic/xp/server/udc/impl/UdcActivator.java  2
- modules/core/core-event/src/main/java/com/enonic/xp/core/impl/event/EventPublisherExecutorImpl.java  1
- modules/core/core-cluster/src/main/java/com/enonic/xp/cluster/impl/ClusterCheckSchedulerImpl.java  1
- modules/core/core-internal/src/main/java/com/enonic/xp/core/internal/concurrent/ManagedExecutor.java  3
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/TaskManagerExecutorImpl.java  1
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/issue/IssueMailSendExecutorImpl.java  1
- modules/core/core-internal/src/test/java/com/enonic/xp/core/internal/concurrent/ManagedExecutorTest.java  1
- modules/core/core-internal/src/test/java/com/enonic/xp/core/internal/concurrent/ManagedRecurringJobSchedulerTest.java  1
- modules/core/core-cluster/src/test/java/com/enonic/xp/cluster/impl/ClusterCheckSchedulerImplTest.java  1
- modules/server/server-trace/src/main/java/com/enonic/xp/server/internal/trace/event/TraceEventDispatcherExecutorImpl.java  1
- modules/server/server-udc/src/main/java/com/enonic/xp/server/udc/impl/UdcSchedulerImpl.java  1
- modules/core/core-internal/src/main/java/com/enonic/xp/core/internal/concurrent/ManagedRecurringJobScheduler.java  4
- modules/core/core-app/src/main/java/com/enonic/xp/core/impl/app/ApplicationListenerHubExecutorImpl.java  1
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/TaskManagerCleanupSchedulerImpl.java  1
         

Complexity decreasing per file
==============================
+ modules/server/server-udc/src/main/java/com/enonic/xp/server/udc/impl/UdcService.java  -1
+ modules/core/core-event/src/main/java/com/enonic/xp/core/impl/event/EventPublisherImpl.java  -1
+ modules/core/core-cluster/src/test/java/com/enonic/xp/cluster/impl/ClusterManagerImplTest.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=4812607&bid=16006411)

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/server/server-udc/src/main/java/com/enonic/xp/server/udc/impl/UdcSchedulerImpl.java  1
- modules/core/core-cluster/src/test/java/com/enonic/xp/cluster/impl/ClusterCheckSchedulerImplTest.java  1
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/TaskManagerCleanupSchedulerImpl.java  1
- modules/core/core-internal/src/main/java/com/enonic/xp/core/internal/concurrent/ManagedExecutor.java  3
- modules/core/core-app/src/main/java/com/enonic/xp/core/impl/app/ApplicationListenerHubExecutorImpl.java  1
- modules/core/core-internal/src/main/java/com/enonic/xp/core/internal/concurrent/ManagedRecurringJobScheduler.java  4
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/issue/IssueMailSendExecutorImpl.java  1
- modules/server/server-trace/src/main/java/com/enonic/xp/server/internal/trace/event/TraceEventDispatcherExecutorImpl.java  1
- modules/core/core-cluster/src/main/java/com/enonic/xp/cluster/impl/ClusterCheckSchedulerImpl.java  1
- modules/core/core-internal/src/main/java/com/enonic/xp/core/internal/concurrent/ThreadFactoryImpl.java  2
- modules/core/core-internal/src/test/java/com/enonic/xp/core/internal/concurrent/ManagedExecutorTest.java  1
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/TaskManagerExecutorImpl.java  1
- modules/server/server-udc/src/main/java/com/enonic/xp/server/udc/impl/UdcActivator.java  2
- modules/core/core-internal/src/test/java/com/enonic/xp/core/internal/concurrent/ManagedRecurringJobSchedulerTest.java  1
- modules/core/core-event/src/main/java/com/enonic/xp/core/impl/event/EventPublisherExecutorImpl.java  1
         

Complexity decreasing per file
==============================
+ modules/core/core-cluster/src/test/java/com/enonic/xp/cluster/impl/ClusterManagerImplTest.java  -1
+ modules/server/server-udc/src/main/java/com/enonic/xp/server/udc/impl/UdcService.java  -1
+ modules/core/core-event/src/main/java/com/enonic/xp/core/impl/event/EventPublisherImpl.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=4812607&bid=16006411)

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/core/core-cluster/src/main/java/com/enonic/xp/cluster/impl/ClusterCheckSchedulerImpl.java  1
- modules/core/core-internal/src/main/java/com/enonic/xp/core/internal/concurrent/ThreadFactoryImpl.java  2
- modules/core/core-internal/src/main/java/com/enonic/xp/core/internal/concurrent/ManagedRecurringJobScheduler.java  4
- modules/core/core-internal/src/test/java/com/enonic/xp/core/internal/concurrent/ManagedExecutorTest.java  1
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/TaskManagerCleanupSchedulerImpl.java  1
- modules/core/core-app/src/main/java/com/enonic/xp/core/impl/app/ApplicationListenerHubExecutorImpl.java  1
- modules/core/core-internal/src/main/java/com/enonic/xp/core/internal/concurrent/ManagedExecutor.java  3
- modules/core/core-event/src/main/java/com/enonic/xp/core/impl/event/EventPublisherExecutorImpl.java  1
- modules/server/server-udc/src/main/java/com/enonic/xp/server/udc/impl/UdcSchedulerImpl.java  1
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/issue/IssueMailSendExecutorImpl.java  1
- modules/server/server-udc/src/main/java/com/enonic/xp/server/udc/impl/UdcActivator.java  2
- modules/core/core-cluster/src/test/java/com/enonic/xp/cluster/impl/ClusterCheckSchedulerImplTest.java  1
- modules/core/core-internal/src/test/java/com/enonic/xp/core/internal/concurrent/ManagedRecurringJobSchedulerTest.java  1
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/TaskManagerExecutorImpl.java  1
- modules/server/server-trace/src/main/java/com/enonic/xp/server/internal/trace/event/TraceEventDispatcherExecutorImpl.java  1
         

Complexity decreasing per file
==============================
+ modules/core/core-event/src/main/java/com/enonic/xp/core/impl/event/EventPublisherImpl.java  -1
+ modules/core/core-cluster/src/test/java/com/enonic/xp/cluster/impl/ClusterManagerImplTest.java  -1
+ modules/server/server-udc/src/main/java/com/enonic/xp/server/udc/impl/UdcService.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=4812607&bid=16006411)

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- modules/core/core-internal/src/main/java/com/enonic/xp/core/internal/concurrent/ThreadFactoryImpl.java  2
- modules/core/core-internal/src/main/java/com/enonic/xp/core/internal/concurrent/SimpleExecutor.java  3
- modules/server/server-udc/src/main/java/com/enonic/xp/server/udc/impl/UdcActivator.java  2
- modules/core/core-internal/src/test/java/com/enonic/xp/core/internal/concurrent/SimpleRecurringJobSchedulerTest.java  1
- modules/core/core-internal/src/test/java/com/enonic/xp/core/internal/concurrent/SimpleExecutorTest.java  1
- modules/core/core-event/src/main/java/com/enonic/xp/core/impl/event/EventPublisherExecutorImpl.java  1
- modules/core/core-cluster/src/test/java/com/enonic/xp/cluster/impl/ClusterCheckSchedulerImplTest.java  1
- modules/core/core-cluster/src/main/java/com/enonic/xp/cluster/impl/ClusterCheckSchedulerImpl.java  1
- modules/core/core-internal/src/main/java/com/enonic/xp/core/internal/concurrent/SimpleRecurringJobScheduler.java  4
- modules/server/server-udc/src/main/java/com/enonic/xp/server/udc/impl/UdcSchedulerImpl.java  1
- modules/server/server-trace/src/main/java/com/enonic/xp/server/internal/trace/event/TraceEventDispatcherExecutorImpl.java  1
- modules/core/core-app/src/main/java/com/enonic/xp/core/impl/app/ApplicationListenerHubExecutorImpl.java  1
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/issue/IssueMailSendExecutorImpl.java  1
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/TaskManagerExecutorImpl.java  1
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/TaskManagerCleanupSchedulerImpl.java  1
         

Complexity decreasing per file
==============================
+ modules/core/core-cluster/src/test/java/com/enonic/xp/cluster/impl/ClusterManagerImplTest.java  -1
+ modules/server/server-udc/src/main/java/com/enonic/xp/server/udc/impl/UdcService.java  -1
+ modules/core/core-event/src/main/java/com/enonic/xp/core/impl/event/EventPublisherImpl.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=4812607&bid=16006411)

File: modules/core/core-internal/src/test/java/com/enonic/xp/core/internal/concurrent/SimpleRecurringJobSchedulerTest.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class SimpleRecurringJobSchedulerTest
+{
+    private static final Duration delay = Duration.ofMillis( 1 );
```
Comment (jsi) [inline modules/core/core-internal/src/test/java/com/enonic/xp/core/internal/concurrent/SimpleRecurringJobSchedulerTest.java]: Codacy Issue found: [Name 'delay' must match pattern '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'.](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=4812607)

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/TaskManagerCleanupSchedulerImpl.java  1
- modules/server/server-trace/src/main/java/com/enonic/xp/server/internal/trace/event/TraceEventDispatcherExecutorImpl.java  1
- modules/server/server-udc/src/main/java/com/enonic/xp/server/udc/impl/UdcSchedulerImpl.java  1
- modules/core/core-app/src/main/java/com/enonic/xp/core/impl/app/ApplicationListenerHubExecutorImpl.java  1
- modules/core/core-internal/src/test/java/com/enonic/xp/core/internal/concurrent/SimpleExecutorTest.java  1
- modules/server/server-udc/src/main/java/com/enonic/xp/server/udc/impl/UdcActivator.java  2
- modules/core/core-cluster/src/test/java/com/enonic/xp/cluster/impl/ClusterCheckSchedulerImplTest.java  1
- modules/core/core-internal/src/main/java/com/enonic/xp/core/internal/concurrent/ThreadFactoryImpl.java  2
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/TaskManagerExecutorImpl.java  1
- modules/core/core-internal/src/main/java/com/enonic/xp/core/internal/concurrent/SimpleRecurringJobScheduler.java  4
- modules/core/core-event/src/main/java/com/enonic/xp/core/impl/event/EventPublisherExecutorImpl.java  1
- modules/core/core-internal/src/test/java/com/enonic/xp/core/internal/concurrent/SimpleRecurringJobSchedulerTest.java  1
- modules/core/core-cluster/src/main/java/com/enonic/xp/cluster/impl/ClusterCheckSchedulerImpl.java  1
- modules/core/core-internal/src/main/java/com/enonic/xp/core/internal/concurrent/SimpleExecutor.java  3
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/issue/IssueMailSendExecutorImpl.java  1
         

Complexity decreasing per file
==============================
+ modules/server/server-udc/src/main/java/com/enonic/xp/server/udc/impl/UdcService.java  -1
+ modules/core/core-cluster/src/test/java/com/enonic/xp/cluster/impl/ClusterManagerImplTest.java  -1
+ modules/server/server-trace/src/main/java/com/enonic/xp/server/internal/trace/event/TraceEventDispatcherImpl.java  -2
+ modules/core/core-event/src/main/java/com/enonic/xp/core/impl/event/EventPublisherImpl.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=4812607&bid=16006411)

**Comment by rymsha** [comment]
> So if I understand correctly, the purpose of "core-internals" is to have a bundle with common classes used by any runtime bundles (incl. admin-impl) but not part of the public API

yes.

**Pull request merged**